feat/order

zi-eunn opened the pull request:
## 개요

<!---- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
1. 주문 기능 구현을 위한 Order, OrderItem, User, Product 엔티티 생성 및 연관관계 설정
2. OrderService를 통해 주문 생성 시 재고 차감 및 예외 처리 로직 작성
3. 주문 요청/응답을 위한 DTO 클래스들 작성
4. 주문 성공/실패에 따른 API 공통 응답 포맷 적용
5. 단위 테스트(OrderServiceTest)를 통한 주문 정상 동작 확인

<!---- Resolves: #(Isuue Number) -->

## PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [x] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)

![스크린샷 2025-06-12 215509](https://github.com/user-attachments/assets/54f21bb5-54ee-451f-93c6-d4ace0bcf024)
![스크린샷 2025-06-12 215518](https://github.com/user-attachments/assets/c94fa528-8567-4885-8edb-11038e6919c8)

## 💬 공유사항 to 리뷰어

<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->

1. 주문 생성 시 상품의 재고 수량을 차감하며, 재고가 부족할 경우 예외 발생(리팩토링 필요 시 CustomException으로 리팩토링)
2. 공통 응답 포맷에 따라 성공/실패 응답을 ApiResponse 구조로 반환
3. 테스트용 유저, 상품 데이터를 직접 생성 후 주문 생성이 정상 동작하는지 OrderServiceTest에서 검증 완료


6. 공통 응답 형식은 아래와 같습니다. 
```
  "success": true,
  "data": {
    "orderId": 1,
    "orderDate": "2025-06-12T12:34:56",
    "totalQuantity": 2
  },
  "error": null
```
실패 예시:
```
  "success": false,
  "data": null,
  "error": {
    "code": "OUT_OF_STOCK",
    "message": "재고가 부족합니다."
  }
```

## PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] Eslint 및 Prettier 규칙을 준수했습니다.
- [x] origin/main에서의 최신 커밋을 확인하고 반영했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

zi-eunn committed:
chore: Lombok 의존성 추가

zi-eunn committed:
feat: 주문(order) 엔티티 생성

- 주문 도메인 구현 : Order entity
- 주문 항목 관리 : OrderItem entity
- 사용자 연관 관계 : User entity
- 상품 연관 관계 : Product entity

zi-eunn committed:
feat: OrderItemDto 생성

zi-eunn committed:
feat: OrderRequestDto 생성

zi-eunn committed:
refactor: OrderItemDto 객체 형태로 리팩토링

zi-eunn committed:
feat: OrderResponseDto 생성, refactor: Order entity 오타 수정

zi-eunn committed:
feat: OrderRepository 생성

zi-eunn committed:
feat: OrderItemRepository 생성

zi-eunn committed:
Merge remote-tracking branch 'origin/develop' into feat/order

zi-eunn committed:
fix: checkstyle 경고 수정 및 LF 줄바꿈 통일

zi-eunn committed:
fix: checkstyle 경고 수정 및 LF 줄바꿈 통일

zi-eunn committed:
fix: checkstyle 경고 수정 및 LF 줄바꿈 통일

zi-eunn committed:
fix: checkstyle 경고 수정 및 LF 줄바꿈 통일

zi-eunn committed:
fix: checkstyle 경고 수정 및 LF 줄바꿈 통일

zi-eunn committed:
fix: checkstyle 경고 수정 및 LF 줄바꿈 통일

zi-eunn committed:
feat: Security DB 접근 허용

zi-eunn committed:
feat: OrderService 생성

zi-eunn committed:
feat: OrderController 구현 및 API 응답 포맷 적용

- feat: OrderController 구현
- chore: ApiResponse, ApiError 생성하여 공통 응답 포맷 반영

zi-eunn committed:
feat: orderservicetest 구현 후 정상 동작 확인

yuj118 reviewed (approved):
늦어서 죄송합니다! 확인했습니다 😊 수고하셨어요

zi-eunn committed:
build: 스프링 부트 버전 3.5.0 -> 3.2.5로 변경

zi-eunn committed:
refector: User Entity 통일(PK 변수형 통일 전)

zi-eunn committed:
chore: 커밋 메시지 변경

zi-eunn committed:
refactor: User Entity 통일

zi-eunn committed:
refactor: Product entity 타입 통일(PK 통일 전)

zi-eunn committed:
chore: build.gradle conflict 수정

zi-eunn committed:
Merge branch 'develop' into feat/order

zi-eunn committed:
refactor: PR#3 ApiResponse에 맞춰 코드 수정

zi-eunn committed:
refactor: Product entity 통일

zi-eunn committed:
feat: 장바구니 기반 주문을 위한 CartItem entity 생성

zi-eunn committed:
feat: 장바구니 기반 주문을 위한 CartItemRepository 생성

zi-eunn committed:
feat: 장바구니 기반 주문 OrderService

zi-eunn committed:
feat: 장바구니 기반 주문 OrderController

zi-eunn committed:
feat: 결제금액 Dto 추가

zi-eunn committed:
feat: 주문 상세 및 목록 조회 API 추가

- Get/orders/{id} : 주문 상세 조회
- Get/orders?email= : 로그인 사용자의 주문 내역 전체 조회
- OrderResponseDto에 총 결제 금액 필드 추가
